File: iris-did/src/main/java/city/smartb/iris/did/deserializer/DIDVerificationMethodDeserializer.java
package city.smartb.iris.did.deserializer;

import city.smartb.iris.did.model.DIDVerificationMethod;
import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.deser.std.StdDeserializer;

import java.io.IOException;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

public class DIDVerificationMethodDeserializer extends StdDeserializer<DIDVerificationMethod> {

    public DIDVerificationMethodDeserializer() { this(null); }

    public DIDVerificationMethodDeserializer(Class<?> vc) {
        super(vc);
    }

    @Override
    public DIDVerificationMethod deserialize(JsonParser jp, DeserializationContext ctxt)
            throws IOException {
        JsonNode node = jp.getCodec().readTree(jp);

        ObjectMapper mapper = new ObjectMapper();

//        List<Object> result = new LinkedHashMap<>();
//        node.forEach( it ->
//                result.
//        );
        Map<String, Object> result = mapper.convertValue(node, new TypeReference<Map<String, Object>>(){});

        return new DIDVerificationMethod(result);
    }
}

File: iris-ld/iris-did/src/main/kotlin/city/smartb/iris/ld/did/serializer/DIDVerificationMethodDeserializer.kt
package city.smartb.iris.ld.did.serializer//package city.smartb.iris.did.model.serializer

import city.smartb.iris.ld.did.DIDVerificationMethod
import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.core.type.TypeReference
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.JsonNode
import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.deser.std.StdDeserializer
import java.io.IOException

class DIDVerificationMethodDeserializer @JvmOverloads constructor(vc: Class<*>? = null) :
    StdDeserializer<DIDVerificationMethod?>(vc) {
    @Throws(IOException::class)

    override fun deserialize(jp: JsonParser, ctxt: DeserializationContext): DIDVerificationMethod {
        val node = jp.codec.readTree<JsonNode>(jp)
        val mapper = ObjectMapper()
        val result: Map<String, Any> =
            mapper.convertValue<Map<String, Any>>(node, object : TypeReference<Map<String, Any>>() {})

        return DIDVerificationMethod(result)
    }
}
